### DISCUSSION TITLE
#5 5th Sprint - Running Demo, Auth, Databaaaaaaaaaaaaase

### DISCUSSION THREAD

**Discussion by AlexMi-Ha** [CUMGroup/AMOGUS#25] [Blog]
Hakuna Matata AMOGUS fans,

A lot of things happened in this sprint and we are excited to show you our progress!
We made huge progress in both the frontend and backend, and finished our ER-Diagram for our Database design. 

For development, we will use SQLite as a database, due to its simplicity.
@Annalytic-programming worked hard on an(n)alyzing our data-storage needs and came up with following Diagram:
![db_concept](https://user-images.githubusercontent.com/61502536/201954727-60b6c525-01ef-4145-a999-c18440420981.png)

All these entities and relationships are already implemented in our Backend using the [EntityFramework](https://learn.microsoft.com/de-de/ef/) Library, which translates the tables and relations into classes, objects and relations (and vice versa). @Annalytic-programming , @AlexMi-Ha and @jas20202 refined the ER-Scheme and implemented the necessary Entities in C#.

Furthermore @AlexMi-Ha managed to implement the AuthService and Controller in our Backend. Take a look:

![swaggerui](https://user-images.githubusercontent.com/61502536/201953971-95df27cf-cfaf-4270-8520-e21576005fa4.png)
![swaggertest](https://user-images.githubusercontent.com/61502536/201953994-b0b612b7-4555-465b-ab7a-e67355cacfc8.png)

All Api Endpoints of our Application will be visible and testable (for devs) in the [Swagger UI](https://swagger.io/).

Additionally, thanks to @WachtelHD, we have our first demo of our landing page UI ready!
![sadwd](https://user-images.githubusercontent.com/61502536/201972669-981c2506-7117-4c22-a8d2-50546a5e7437.gif)

But that's it for this weeks blog entry.
I hope you are as excited as we are for the upcomming weeks!

Yours truly,
CUMGroup

**Comment by EhrlerL** [comment]
G'day CUMGroup!
Great job on the ER-Diagram! Seems like you already have a precise plan for your database. You definitely didn't promise too much in your last blog entry.
I honestly have to say, I neither used EntityFramework nor Swagger (the former was a bit overwhelming when I looked up the documentation). But afaik both are very useful tools and I bet you'll make the very best of it. 'Til now, the AMOGUS.Api with its authentification looks good. 
The landing page demo is extraordinarily sus. Over all, good job as always and I'm already excited for future updates!
Cheers, capyclue. ₍ᐢ•(ܫ)•ᐢ₎

**Reply by AlexMi-Ha** [reply]
Hello Team Capyclue, 👀
thanks for the recurring feedback, we appreciate a lot!
You are right; if you have never worked with EntityFramework (or similar Object-Relational-Mapping frameworks) it can seem a bit overwhelming in the start. But with a bit of patience it is a really convenient way of managing your database.
For the landing page we tried to add EXTRA sus flavor to achieve a good first impression, when visiting our website!
Yours truly,
CUMGroup

**Comment by willizielke** [comment]
Hi Team AMOGUS,

We really enjoyed your blog post. We like your ER-Diagramm, it is very clear to look at, the important connections are very simple yet good structured. Also, you've already implemented it, with a good explanation of how you did it, well done!
It is also nice to see that your demo already works.  
We hope to see more from you in the upcomming weeks!

Sincerely,
Team RuleTheGym

**Reply by AlexMi-Ha** [reply]
How's it going Team RuleTheGym,
thanks for the positive feedback! Good to hear that our diagrams and explanations are comprehensible. ^^

Yours truly,
CUMGroup
